src/Microsoft.DocAsCode.Build.Engine/SingleDocumentBuilder.cs

```diff
@@ -580,8 +580,12 @@ private void FeedXRefMap(List<ManifestItemWithContext> manifest, IDocumentBuildC
                 using (new LoggerFileScope(m.FileModel.LocalPathFromRepoRoot))
                 {
                     Logger.LogDiagnostic($"Feed xref map from template for {m.Item.DocumentType}...");
-                    var bookmarks = m.Options.Bookmarks;
-                    // TODO: Add bookmarks to xref
+                    // TODO: use m.Options.Bookmarks directly after all templates report bookmarks
+                    var bookmarks = m.Options.Bookmarks ?? m.FileModel.Bookmarks;
```

Comment:
The `FeedOptions` above calls `exports.getOptions` in js.

---
In reply to: [86703507](https://github.com/dotnet/docfx/pull/860#discussion_r86703507) [](ancestors = 86703507,86513579)